src/coreclr/System.Private.CoreLib/src/System/Buffer.CoreCLR.cs

```diff
@@ -16,7 +17,13 @@ public partial class Buffer
         private static unsafe partial void __Memmove(byte* dest, byte* src, nuint len);
 
         [MethodImpl(MethodImplOptions.InternalCall)]
-        private static extern void __BulkMoveWithWriteBarrier(ref byte destination, ref byte source, nuint byteCount);
+        private static extern void BulkMoveWithWriteBarrierInternal(ref byte destination, ref byte source, nuint byteCount);
+
+        private static void __BulkMoveWithWriteBarrier(ref byte destination, ref byte source, nuint byteCount)
+        {
+            Thread.PollGC();
```

Comment:
Last I remember, we call this routine only for small buffers, for large one we call QCall

Reply:
That isn't how this is implemented in the FCall. I'm looking to match behavior at first and I need to understand the performance impact before I start mucking with how these can be changed.